### PR TITLE
F#344 cpp library to detect hand

### PR DESCRIPTION
**Pull request by VahidAminZ**
#344

### CODE REVIEW

**Comment by VahidAminZ** [comment]
Do not merge yet

File: sr_utilities/CMakeLists.txt [hunk 1]
```diff
@@ -33,6 +33,10 @@ catkin_add_gtest(test_math_utils test/test_math_utils.cpp)
 add_dependencies(test_math_utils sr_utilities_gencpp ${catkin_EXPORTED_TARGETS})
 target_link_libraries(test_math_utils ${catkin_LIBRARIES} pthread)
 
+add_rostest_gtest(test_hand_finder test/test_hand_finder.test test/test_hand_finder.cpp)
+add_dependencies(test_hand_finder sr_utilities_gencpp ${catkin_EXPORTED_TARGETS})
+target_link_libraries(test_hand_finder sr_hand_finder ${catkin_LIBRARIES})
```
Comment (toliver) [inline sr_utilities/CMakeLists.txt]: These 3 lines should be inside the `if(CATKIN_ENABLE_TESTING)`. Isn't it?

Comment (VahidAminZ) [inline sr_utilities/CMakeLists.txt]: Yes yes.

File: sr_utilities/CMakeLists.txt [hunk 2]
```diff
@@ -32,7 +32,11 @@ target_link_libraries(test_thread_safe_map ${catkin_LIBRARIES} pthread)
 catkin_add_gtest(test_math_utils test/test_math_utils.cpp)
 add_dependencies(test_math_utils sr_utilities_gencpp ${catkin_EXPORTED_TARGETS})
 target_link_libraries(test_math_utils ${catkin_LIBRARIES} pthread)
-
+if(CATKIN_ENABLE_TESTING)
+add_rostest_gtest(test_hand_finder test/test_hand_finder.test test/test_hand_finder.cpp)
```
Comment (toliver) [inline sr_utilities/CMakeLists.txt]: Indentation?

Comment (VahidAminZ) [inline sr_utilities/CMakeLists.txt]: fixed it

**Comment by VahidAminZ** [comment]
@toliver : fixed the indentation

**Comment by toliver** [comment]
Could you try what happens if no hand is present? (no "hand/mapping" or joint prefix parameters present)

**Comment by VahidAminZ** [comment]
@toliver It reads the parameters from the description file but it will crash if no joint prefix is present.

**Comment by toliver** [comment]
That would need to be fixed. The idea behind the library is that a node will use it to discover what hands are present. So if no hands are present it doesn't have to crash. It should return empty maps or something similar.

**Comment by VahidAminZ** [comment]
@toliver : I think I got confused between this one and other PR for fixing the tests. It should be fine to run it without any hand. I will write a test for it.

**Comment by VahidAminZ** [comment]
@toliver added the extra test

**Pull request merged**